packages/eslint-plugin/src/rules/naming-convention.ts

```diff
@@ -1309,16 +1314,23 @@ function normalizeOption(option: Selector): NormalizedSelector {
     ? option.selector
     : [option.selector];
 
-  return {
-    selector: selectors
-      .map(selector =>
-        isMetaSelector(selector)
-          ? MetaSelectors[selector]
-          : Selectors[selector],
-      )
-      .reduce((accumulator, selector) => accumulator | selector),
-    ...normalizedOption,
-  };
+  const allowedType = [1 << 0, 1 << 2, 1 << 3, 1 << 4, 1 << 6];
+  const config: NormalizedSelector[] = [];
+  selectors
+    .map(selector =>
+      isMetaSelector(selector) ? MetaSelectors[selector] : Selectors[selector],
+    )
+    .forEach(selector =>
+      allowedType.includes(selector)
```

Comment:
nit - use the selector enum instead of redefining the numbers.

```suggestion
  const selectorsAllowedToHaveTypes = new Set<Selector>([
    Selectors.variable,
    Selectors.parameter,
    Selectors.property,
    Selectors.parameterProperty,
    Selectors.accessor,
  ]);
  const config: NormalizedSelector[] = [];
  selectors
    .map(selector =>
      isMetaSelector(selector) ? MetaSelectors[selector] : Selectors[selector],
    )
    .forEach(selector =>
      selectorsAllowedToHaveTypes.has(selector)
```